components/tyk-dev-portal/templates/job-portal.yaml

```diff
@@ -0,0 +1,56 @@
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: dev-portal-job-{{ include "tyk-dev-portal.fullname" . }}
+  labels:
+    app: {{ include "tyk-dev-portal.fullname" . }}
+    chart: {{ include "tyk-dev-portal.chart" . }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}  
+spec:
+  template:
+    spec:
+      containers:
+      - name: bootstrap-portal
+        image: curlimages/curl:8.8.0
+        env:
+        - name: ADMIN_EMAIL
+          value: "{{ .Values.global.adminUser.email  }}"
+        - name: ADMIN_PASSWORD
+          {{ if .Values.global.adminUser.useSecretName }}
+          valueFrom:
+            secretKeyRef:
+              name: secrets-{{ include "tyk-dev-portal.fullname" . }}
+              key: adminUserPassword
+          {{ else }}
+          value: {{ .Values.global.adminUser.password | quote }}
+          {{ end }}
+        - name: FIRSTNAME
+          value: "{{ .Values.bootstrapPortal.firstName }}"
+        - name: LASTNAME
+          value: "{{ .Values.bootstrapPortal.lastName }}"                    
+        - name: SERVICE_NAME
+          value: "dev-portal-svc-{{ include "tyk-dev-portal.fullname" . }}.{{ .Release.Namespace }}.svc.cluster.local"
+        - name: PORT
+          value: "{{ .Values.containerPort }}"   
+        command: 
+          - sh
+          - -c
+          - |
+            echo "Waiting for dev-portal to be ready..."
+            while ! nc -z $SERVICE_NAME $PORT; do
+              echo "dev-portal is not ready yet. Waiting..."
+              sleep 1
+            done
+            echo "dev-portal is ready. Making the API call..."
+            curl --silent --show-error --fail --location --request POST "http://$SERVICE_NAME:$PORT/portal-api/bootstrap" \
```

Comment:
do we need to bootstrap portal through HTTP request instead of `bootstrap` command of it, as we do here: https://github.com/TykTechnologies/tyk-charts/blob/48b6ad29709055e6d0299d504b1b41a9f404d473/components/tyk-dev-portal/templates/statefulset-enterprise-portal.yaml#L178-L180?

If there is no diff between using CLI and HTTP request, maybe we can use `tyk-dev-portal` image and run `bootstrap` command here instead of writing custom handler.

If we have to bootstrap portal through HTTP requests, maybe it would be better to implement this logic in https://github.com/TykTechnologies/tyk-k8s-bootstrap? What do you think

Reply:
I think the command relies on a secret which won't be created until after the post script hooks runs, with API, the dev portal does not need any secret and Moreover, the requirement was that it should be done via HTTP.